### ISSUE TITLE
Exception when opening a fit for a ship with no slots

### ISSUE THREAD

**Issue by m-sasha** [pyfa-org/Pyfa#2241]
If I open a fit for a shuttle, PyFa shows this error:

```
pyfa v2.23.0
EVE Data Version: 1769230 (2020-07-14 13:49:15)

OS version: Windows-10-10.0.17134-SP0
Python version: 3.7.8 (tags/v3.7.8:4b47a5b6ba, Jun 28 2020, 08:53:46) [MSC v.1916 64 bit (AMD64)]
wxPython version: 4.0.6 (wxWidgets 3.0.5)
SQLAlchemy version: 1.3.18
Logbook version: 1.5.3
Requests version: 2.24.0
Dateutil version: 2.8.1

####################

Traceback (most recent call last):
  File "gui\builtinViews\fittingView.py", line 91, in fitSelected
  File "gui\builtinViews\fittingView.py", line 346, in fitSelected
  File "gui\builtinViews\fittingView.py", line 591, in slotsChanged
  File "gui\builtinViews\fittingView.py", line 565, in generateMods
IndexError: list index out of range
```

**Comment by blitzmann** [comment]
Is this is new fit or one that you already had?

Can you post your log files? Specifically the one related to the error. They can be found `%userprofile%/.pyfa/*.log`

**Comment by m-sasha** [comment]
It happens whether it's a new or an existing fit.
Here's the last part of the log:
```
[2020-07-28 19:48:14.213373] DEBUG: gui.builtinViews.fittingView: ------------------ new fitting view -------------------
[2020-07-28 19:48:14.213373] DEBUG: gui.builtinViews.fittingView: <gui.builtinViews.fittingView.FittingView object at 0x000001A480F77318>
[2020-07-28 19:48:14.213373] DEBUG: gui.builtinViews.fittingView: ###################### Created new view:<gui.builtinViews.fittingView.FittingView object at 0x000001A480F77318>
[2020-07-28 19:48:14.214861] DEBUG: gui.builtinViews.fittingView: ====== Fit Selected: <gui.builtinViews.fittingView.FittingView object at 0x000001A480F77318>True
[2020-07-28 19:48:14.219848] INFO: service.fit: ==========recalc: Gallente Shuttle fit==========
[2020-07-28 19:48:14.222331] INFO: eos.saveddata.fit: Starting fit calculation on: Fit(ID=400, ship=Gallente Shuttle, name=Gallente Shuttle fit) at 0x1a481c1f508, calc: LOCAL
[2020-07-28 19:48:14.223320] INFO: eos.saveddata.fit: Fit is not yet calculated; will be running local calcs for Fit(ID=400, ship=Gallente Shuttle, name=Gallente Shuttle fit) at 0x1a481c1f508
[2020-07-28 19:48:14.230236] DEBUG: eos.saveddata.fit: Done with fit calculation
[2020-07-28 19:48:14.230236] INFO: service.fit: ==========recalc time: 0.010388612747192383==========
[2020-07-28 19:48:14.231724] DEBUG: service.fit: Check states for fit ID: Gallente Shuttle fit (Gallente Shuttle)
[2020-07-28 19:48:14.236684] CRITICAL: gui.errorDialog: 

Traceback (most recent call last):
  File "gui\builtinViews\fittingView.py", line 91, in fitSelected
  File "gui\builtinViews\fittingView.py", line 346, in fitSelected
  File "gui\builtinViews\fittingView.py", line 591, in slotsChanged
  File "gui\builtinViews\fittingView.py", line 565, in generateMods
IndexError: list index out of range
```

**Comment by DarkFenX** [comment]
What are your settings - separate racks, show rack labels, color fitting view by slot?

**Comment by DarkFenX** [comment]
Got the gist of it, you have to disable rack labels and then it reproduces.

**Comment by DarkFenX** [comment]
Fixed in f91e0b2e2326d0136184ae77bf02c5819316b396
